tests/OpenCFP/Http/Controller/Admin/TalksControllerTests.php

```diff
@@ -0,0 +1,184 @@
+        ];
+
+        $talkData = [0 => [
+            'id' => 1,
+            'title' => 'Test Title',
+            'description' => "The title should contain this & that",
+            'type' => 'regular',
+            'level' => 'entry',
+            'category' => 'other',
+            'desired' => 0,
+            'slides' => '',
+            'other' => '',
+            'sponsor' => '',
+            'user_id' => $userId,
+            'created_at' => date('Y-m-d'),
+            'user' => $userDetails,
+            'favorite' => null,
+            'selected' => null
+        ]];
+        $userMapper = m::mock('OpenCFP\Domain\Entity\Mapper\User');
+        $userMapper->shouldReceive('migrate');
+        $userMapper->shouldReceive('build')->andReturn($this->app['user']);
+        $userMapper->shouldReceive('save')->andReturn(true);
+
+        $talkMapper = m::mock('OpenCFP\Domain\Entity\Mapper\Talk');
+        $talkMapper->shouldReceive('migrate');
+        $talkMapper->shouldReceive('build')->andReturn($talk);
+        $talkMapper->shouldReceive('save');
+        $talkMapper->shouldReceive('getAllPagerFormatted')->andReturn($talkData);
+
+        // Overide our DB mappers to return doubles
+        $spot = m::mock('Spot\Locator');
+        $spot->shouldReceive('mapper')
+            ->with('OpenCFP\Domain\Entity\User')
+            ->andReturn($userMapper);
+        $spot->shouldReceive('mapper')
+            ->with('OpenCFP\Domain\Entity\Talk')
+            ->andReturn($talkMapper);
+        $this->app['spot'] = $spot;
+
+        $req = m::mock('Symfony\Component\HttpFoundation\Request');
+        $paramBag = m::mock('Symfony\Component\HttpFoundation\ParameterBag');
+
+        $queryParams = [
+            'page' => 1,
+            'per_page' => 20,
+            'sort' => 'ASC',
+            'order_by' => 'title',
+            'filter' => null,
+        ];
+        $paramBag->shouldReceive('all')->andReturn($queryParams);
+
+        $req->shouldReceive('get')->with('page')->andReturn($queryParams['page']);
+        $req->shouldReceive('get')->with('per_page')->andReturn($queryParams['per_page']);
+        $req->shouldReceive('get')->with('sort')->andReturn($queryParams['sort']);
+        $req->shouldReceive('get')->with('order_by')->andReturn($queryParams['order_by']);
+        $req->shouldReceive('get')->with('filter')->andReturn($queryParams['filter']);
+        $req->query = $paramBag;
+        $req->shouldReceive('getRequestUri')->andReturn('foo');
+
+        $this->app['twig']
+            ->addGlobal(
+                'user_is_admin',
+                $this->app['sentry']->getUser()->hasAccess('admin')
+            );
+
+        ob_start();
+        $this->app->run();
+        ob_end_clean();
+
+        $controller = new \OpenCFP\Http\Controller\Admin\TalksController();
+        $controller->setApplication($this->app);
+        $response = $controller->indexAction($req);
+        $this->assertContains('Test Title', (string) $response);
```

Comment:
I think there's value in cleaning up tests like this to introduce some level of abstraction that more clearly defines context. This could include factories of any sort, better test helpers, etc. More of a general "improve readability of tests" effort